### PR TITLE
chore:improved github actions to use a matrix

### LINKED ISSUE #1078: Use python 3.9 in GitHub actions.
### Is your feature request related to a problem? Please describe.

Currently we run github actions on python 3.6.

### Describe the solution you'd like

Use a build matrix to run on version 3.7 , 3.9 (maybe `python:latest` for future releases?)

### Additional context

https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix

### PR DESCRIPTION
**Pull request by vj-codes**
### Description

Improved github actions to use a matrix

Fixes #1078 

### Type of Change:

- Code
- Quality Assurance

### Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**

- [ ] My changes generate no new warnings 
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by epicadk** [comment]
Lint check failing. I know you haven't made any changes but can you still run black?

**Comment by vj-codes** [comment]
@epicadk yup see [this comment](https://github.com/anitab-org/mentorship-backend/pull/1074#issuecomment-826442201)
it's passing locally for me but actions is failing

**Comment by vj-codes** [comment]
@epicadk the tests are still running twice , should the changes be done here as the same file is being updated?

**Review by epicadk — DISMISSED** [review]
👍🏼 looks good.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -20,12 +20,16 @@ jobs:
   build:
     needs: lint
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.7, 3.9]
```
Comment (epicadk) [inline .github/workflows/main.yml:25]: Is it not possible to use `latest` for a matrix?

Comment (vj-codes) [reply .github/workflows/main.yml:25]: AFAIK it only accepts numerical values but giving a try with `latest`

Comment (vj-codes) [reply .github/workflows/main.yml:25]: @epicadk it isn't possible , will revert the changes

**Comment by epicadk** [comment]
Ready to merge after one more review.

**Review by epicadk — APPROVED** [review]

**Review by isabelcosta — APPROVED** [review]
Great work @vj-codes 🎉 
Thank you so much for working on this!

**Pull request merged**